### PR TITLE
Adds additional information about the stringifier IDL member

### PR DESCRIPTION
**Pull request by jpmedley**
The information about `stringifier` is incomplete. There places in the IDL spec that describe what this does.

https://heycam.github.io/webidl/#idl-stringifiers
https://heycam.github.io/webidl/#es-stringifier

### CODE REVIEW

File: files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html [hunk 1]
```diff
@@ -12,19 +12,19 @@
...
-   <li>Similarly, WebCore, as expected because Chromium was forked from it, also scatters them: E.g. <a href="https://github.com/WebKit/webkit/blob/master/Source/WebCore/html/DOMTokenList.idl">https://github.com/WebKit/webkit/blob/master/Source/WebCore/html/DOMTokenList.idl</a></li>
+   <li>For Chromium, they are located in two locations, both subtrees of the source code's <code><a href="https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/">renderer/<a></a></code> directory: <code><a href="https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/core/">core/</a></code> and <code><a href="https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/modules/">modules/</a></code>. Chromium source code has IDL files in other locations, but these are part of the testing system and not relevant to API implementations.</li>
+   <li>For WebCore, are scattered around the source code, so you need to dig a bit more: E.g. <a href="https://github.com/WebKit/webkit/blob/master/Source/WebCore/html/DOMTokenList.idl">https://github.com/WebKit/webkit/blob/master/Source/WebCore/html/DOMTokenList.idl</a></li>
```
Comment (sideshowbarker) [inline files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html:27]:
```suggestion
   <li>For WebCore, they are scattered around the source code, so you need to dig a bit more: E.g. <a href="https://github.com/WebKit/webkit/blob/master/Source/WebCore/html/DOMTokenList.idl">https://github.com/WebKit/webkit/blob/master/Source/WebCore/html/DOMTokenList.idl</a></li>
```

File: files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html [hunk 2]
```diff
@@ -485,6 +485,34 @@ <h3 id="Set-like_methods">Set-like methods</h3>
 
 <p>Such an set interface also allows to use the syntax <code>for (var p in object)</code> as a shorthand of <code>for (var p in object.entries())</code>. We add a sentence about it in the interface description. E.g. {{domxref('FontFaceSet')}}.</p>
 
+<h2 id="Special_Behaviors">Special Behaviors</h2>
+
+<p>Some IDL members indicate special behaviors that should be noted on appropriate pages.</p>
+
+<h3 id="Stringifiers">Stringifiers</h3>
+
+<p>In addition to adding the <code>toString()</code> method to an interface as described in <a href="#toString_and_toJSON">toString() and toJason()</a>, stringifiers also indicate that an object instance, when used as a string, returns a <code>DOMString</code> other than the default (usually a JSON representation of the object). Exactly how depends on the way it is specified in the IDL.</p>
```
Comment (sideshowbarker) [inline files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html:494]:
```suggestion
<p>In addition to adding the <code>toString()</code> method to an interface as described in <a href="#toString_and_toJSON">toString() and toJSON()</a>, stringifiers also indicate that an object instance, when used as a string, returns a <code>DOMString</code> other than the default (usually a JSON representation of the object). Exactly how depends on the way it is specified in the IDL.</p>
```

File: files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html [hunk 3]
```diff
@@ -485,6 +485,34 @@ <h3 id="Set-like_methods">Set-like methods</h3>
...
+
+<p>When stringifier accompanies an attribute name, referecing the object name has the same result as referencing the attribute name. Consider the following IDL:</p>
```
Comment (sideshowbarker) [inline files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html:496]:
```suggestion
<p>When a stringifier accompanies an attribute name, referencing the object name has the same result as referencing the attribute name. Consider the following IDL:</p>
```

**Review by sideshowbarker — CHANGES_REQUESTED** [review]

File: files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html (hunk 2 above)
Comment (jpmedley) [reply files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html:494]: LOL.

**Comment by jpmedley** [comment]
@sideshowbarker Apologies for requesting a second review. After looking at what I did with fresh eyes this morning, I realized that what I submitted wasn't quite complete. My fixes are here: https://github.com/mdn/content/pull/270/commits/d57a032d1af96bd01c6b9eace12842c78dc01ce8

The misspelling of `toJSON()` is corrected in the final version even though it doesn't show it here. I forgot to pull before I started making local changes. The correction is in the merge commit.

File: files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html [hunk 4]
```diff
@@ -392,7 +392,7 @@ <h2 id="Special_methods">Special methods</h2>
 
 <h3 id="toString_and_toJSON">toString() and toJSON()</h3>
 
-<p>A stringifier is mapped to <code>toString()</code> and defined as:</p>
+<p>A stringifier specifies how an object based on an interface is resolved in contexts expecting a string. (See <a href="Stringifiers">stringifiers</a>.) Additionaly, the keyword is mapped to <code>toString()</code> and defined as:</p>
```
Comment (sideshowbarker) [inline files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html:395]:
```suggestion
<p>A stringifier that specifies how an object based on an interface is resolved in contexts expecting a string. (See <a href="Stringifiers">stringifiers</a>.) Additionaly, the keyword is mapped to <code>toString()</code> and defined as:</p>
```

File: files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html [hunk 5]
```diff
@@ -485,6 +485,34 @@ <h3 id="Set-like_methods">Set-like methods</h3>
...
 
+<h2 id="Special_Behaviors">Special Behaviors</h2>
+
+<p>Some IDL members indicate special behaviors that should be noted on appropriate pages.</p>
+
+<h3 id="Stringifiers">Stringifiers</h3>
+
+<p>In addition to adding the <code>toString()</code> method to an interface as described in <a href="#toString_and_toJSON">toString() and toJSON()</a>, stringifiers also indicate that an object instance, when used as a string, returns a <code>DOMString</code> other than the default. (The default is usually a JSON representation of the object). Exactly how depends on the way it is specified in the IDL. Regardless of the how, the non-default behavior should be described on the interface page.</p>
+
+<p>When stringifier accompanies an attribute name, referencing the object name has the same result as referencing the attribute name. Consider the following IDL:</p>
```
Comment (sideshowbarker) [inline files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html:496]:
```suggestion
<p>When the <code>stringifier</code> keyword accompanies an attribute name, referencing the object name has the same result as referencing the attribute name. Consider the following IDL:</p>
```

File: files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html [hunk 6]
```diff
@@ -485,6 +485,34 @@ <h3 id="Set-like_methods">Set-like methods</h3>
...
+  stringifier DOMString name();
+};</span></pre>
+
+<p>For a class based on this interface, the following lines of code below are equivalent. The behavior should be noted on the property page in addition to the interface page.</p>
+
+<pre class="js" lang="en"><span id="the-code">console.log(interfaceIdentifier);
+console.log(interfaceIdentifier.name);
+</span></pre>
+
+<p>When the stringifier keyword is used by itself, an object of the interface may be used as above, but the behavior is defined in source code.</p>
```
Comment (sideshowbarker) [inline files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html:508]:
```suggestion
<p>When the <code>stringifier</code> keyword is used by itself, an object of the interface may be used as above, but the behavior is defined in source code.</p>
```

File: files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html [hunk 7]
```diff
@@ -485,6 +485,34 @@ <h3 id="Set-like_methods">Set-like methods</h3>
...
+<p>Some IDL members indicate special behaviors that should be noted on appropriate pages.</p>
+
+<h3 id="Stringifiers">Stringifiers</h3>
+
+<p>In addition to adding the <code>toString()</code> method to an interface as described in <a href="#toString_and_toJSON">toString() and toJSON()</a>, stringifiers also indicate that an object instance, when used as a string, returns a <code>DOMString</code> other than the default. (The default is usually a JSON representation of the object). Exactly how depends on the way it is specified in the IDL. Regardless of the how, the non-default behavior should be described on the interface page.</p>
+
+<p>When stringifier accompanies an attribute name, referencing the object name has the same result as referencing the attribute name. Consider the following IDL:</p>
+
+<pre class="syntaxbox" lang="en"><span id="the-code">interface InterfaceIdentifier {
+  stringifier DOMString name();
```
Comment (sideshowbarker) [inline files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html:499]:
```suggestion
  stringifier attribute DOMString DOMString name;
```

File: files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html [hunk 8]
```diff
@@ -485,6 +485,34 @@ <h3 id="Set-like_methods">Set-like methods</h3>
...
+<p>For a class based on this interface, the following lines of code below are equivalent. The behavior should be noted on the property page in addition to the interface page.</p>
+
+<pre class="js" lang="en"><span id="the-code">console.log(interfaceIdentifier);
+console.log(interfaceIdentifier.name);
+</span></pre>
+
+<p>When the stringifier keyword is used by itself, an object of the interface may be used as above, but the behavior is defined in source code.</p>
+
+<pre class="syntaxbox" lang="en"><span id="the-code">interface InterfaceIdentifier {
+  stringifier DOMString name();
```
Comment (sideshowbarker) [inline files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html:511]:
```suggestion
  stringifier;
```

**Review by sideshowbarker — CHANGES_REQUESTED** [review]

File: files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html (hunk 4 above)
Comment (jpmedley) [reply files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html:395]: That would turn a sentence into a clause. Might check that request again.

**Comment by jpmedley** [comment]
Please take another look.

File: files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html (hunk 4 above)
Comment (sideshowbarker) [reply files/en-us/mdn/contribute/howto/write_an_api_reference/information_contained_in_a_webidl_file/index.html:395]: > That would turn a sentence into a clause. Might check that request again.

Yup, I see now

**Review by sideshowbarker — APPROVED** [review]

**Comment by sideshowbarker** [comment]
LGTM but I don’t have perms to merge it, so I guess we need @Elchi3 or @chrisdavidmills to do the honors

**Review by chrisdavidmills — APPROVED** [review]
I'll make it so.

**Pull request merged**